ek9000App/src/devEK9000.cpp

```diff
@@ -285,30 +280,32 @@ int devEK9000Terminal::getEK9000IO(int type, int startaddr, uint16_t* buf, int l
 //		Holds useful vars for interacting with EK9000/EL****
 //		hardware
 //==========================================================//
-devEK9000::devEK9000() {
-	/* Initialize members*/
-	m_terms = NULL;
-	m_numTerms = 0;
-	m_driver = NULL;
+devEK9000::devEK9000(const char* portName, const char* octetPortName, int termCount, const char* ip)
+	: drvModbusAsyn(portName, octetPortName, 0, 2, -1, 256, dataTypeUInt16, 150, "") {
+
+	/* Initialize members */
+	m_terms = static_cast<devEK9000Terminal*>(malloc(sizeof(devEK9000Terminal) * termCount));
```

Comment:
For my own information, why the mix of malloc/free and not just use `new[]` operator to allocate all terminals?

Reply:
I was confused at my own code here too, but I just remembered: `new[]` requires a default constructible type, which `devEK9000Terminal` is not because its constructor takes a parameter.

Reply:
I'll probably make `m_terms` a `std::vector<devEK9000Terminal*>` instead- just realized that the destructor is not being called on any `devEK9000Terminal`s because the array is simply being freed.

Reply:
Ah, I see. I like the `std::vector` approach much better 👍